dataland-frontend/tests/e2e/fixtures/frameworks/lksg/LksgPreparedFixtures.ts

```diff
@@ -1,187 +1,182 @@
 
 /**
- * Generates LkSG prepared fixtures by generating random LkSG datasets and afterwards manipulating some fields
- * via manipulator-functions to set specific values for those fields.
+ * Generates lksg prepared fixtures by generating random lksg datasets and
+ * afterward manipulating some fields via manipulator-functions to set specific values for those fields.
  * @returns the prepared fixtures
  */
 export function generateLksgPreparedFixtures(): Array<FixtureData<LksgData>> {
   const preparedFixtures = [];
-  preparedFixtures.push(manipulateFixtureForSixLksgDataSetsInDifferentYears(generateLksgFixtures(1)[0]));
-  preparedFixtures.push(manipulateFixtureForOneLksgDataSetWithProductionSites(generateLksgFixtures(1, 0)[0]));
-  preparedFixtures.push(manipulateFixtureForAllFields(generateLksgFixtures(1, 0)[0]));
-  preparedFixtures.push(manipulateFixtureForDate(generateLksgFixtures(1)[0], "2023-04-18"));
-  preparedFixtures.push(manipulateFixtureForDate(generateLksgFixtures(1)[0], "2023-06-22"));
-  preparedFixtures.push(manipulateFixtureForDate(generateLksgFixtures(1)[0], "2022-07-30"));
-  preparedFixtures.push(manipulateFixtureForLksgDatasetWithLotsOfNulls(generateOneLksgFixtureWithManyNulls()));
-  preparedFixtures.push(manipulateFixtureToContainProcurementCategories(generateLksgFixtures(1, 0)[0]));
-  preparedFixtures.push(manipulateFixtureToNotBeAManufacturingCompany(generateLksgFixtures(1, 0)[0]));
-  preparedFixtures.push(manipulateFixtureToHaveNoChildLaborUnder18AndChildLaborUnder15(generateLksgFixtures(1, 0)[0]));
+  // Note: Put the code for prepared fixture generation below. This file will not be overwritten automatically
+
+  const manipulatorFunctions: Array<(input: FixtureData<LksgData>) => FixtureData<LksgData>> = [
+    generateFixutreWithNoNullFields,
+    generateFixtureWithALotOfNullFields,
+    generateFixtureToNotBeAManufacturingCompany,
+    generateFixtureToHaveNoChildLaborUnder18AndChildLaborUnder15,
+    generateFixtureToContainProcurementCategories,
+    generateFixtureForSixLksgDataSetsInDifferentYears,
+    generateOneLksgDatasetWithOnlyNulls,
+    generateFixtureForSixLksgDataSetsInDifferentYears,
+    generateFixtureForOneLksgDataSetWithProductionSites,
+  ];
+  const preparedFixturesBeforeManipulation = generateLksgFixtures(manipulatorFunctions.length);
+
+  for (let i = 0; i < manipulatorFunctions.length; i++) {
+    preparedFixtures.push(manipulatorFunctions[i](preparedFixturesBeforeManipulation[i]));
+  }
+  preparedFixtures.push(generateFixtureForDate("2023-04-18"));
+  preparedFixtures.push(generateFixtureForDate("2023-06-22"));
+  preparedFixtures.push(generateFixtureForDate("2022-07-30"));
   return preparedFixtures;
 }
 
+/**
+ * Generates a lksg fixture with no null values
+ * @returns the fixture
+ */
+function generateFixutreWithNoNullFields(): FixtureData<LksgData> {
+  const newFixture = generateLksgFixtures(1, 0)[0];
+  newFixture.companyInformation.companyName = "lksg-all-fields";
```

Comment:
TODO for me: validate that actually all fields are populated. In the previous version there were fields required to be set manually to guarantee that.